### PR TITLE
Remove photon cloud provider and volume plugin

### PR DESCRIPTION
**Pull request by dougm**
vmware/photon-controller is no longer maintained, as of Oct 2017.

See also: PR #58096

**What this PR does / why we need it**:

As the Photon Controller project is no longer maintained, removing the photon cloud provider and volume plugin reduce in-tree maintenance when related interfaces are changed.

**Special notes for your reviewer**:

Photon controller support has already been removed from kube-up in PR #58096

**Release note**:

```release-note
VMware Photon cloud provider and volume plugin have been removed.  The Photon Controller project is no longer maintained as of Oct 2017.
```

### CODE REVIEW

**Comment by dims** [comment]
/approve

(in principle!)

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -136,8 +136,6 @@ type VolumeSource struct {
 	// AzureDisk represents an Azure Data Disk mount on the host and bind mount to the pod.
 	// +optional
 	AzureDisk *AzureDiskVolumeSource
-	// PhotonPersistentDisk represents a Photon Controller persistent disk attached and mounted on kubelets host machine
-	PhotonPersistentDisk *PhotonPersistentDiskVolumeSource
```
Comment (msau42) [inline pkg/apis/core/types.go]: I don't think we can actually remove fields from the API.  The best we can do is mark it as deprecated...

**Comment by dougm** [comment]
Making use of the new deprecation mechanism for now: https://github.com/kubernetes/kubernetes/pull/63686

We can look to remove after the 1.11 release.

**Pull request closed without merging**